File: algorithms/0012-integer-to-roman.py
class Solution:
    def intToRoman(self, num: int) -> str:
        r = str()

        c, num = num // 1000, num % 1000
        r += 'M' * c

        c, num = num // 100, num % 100
        if c == 9:
            r += 'CM'
        elif c in range(5, 9):
            r += 'D' + 'C' * (c - 5)
        elif c == 4:
            r += 'CD'
        else:
            r += 'C' * c

        c, num = num // 10, num % 10
        if c == 9:
            r += 'XC'
        elif c in range(5, 9):
            r += 'L' + 'X' * (c - 5)
        elif c == 4:
            r += 'XL'
        else:
            r += 'X' * c

        c = num
        if c == 9:
            r += 'IX'
        elif c in range(5, 9):
            r += 'V' + 'I' * (c - 5)
        elif c == 4:
            r += 'IV'
        else:
            r += 'I' * c

        return r

File: algorithms/0012-integer-to-roman.go
func intToRoman(num int) string {
	var r string
	var c int

	c, num = num / 1000, num % 1000
	for i := 0; i < c; i++ {
		r += "M"
	}

	c, num = num / 100, num % 100
	switch c {
	case 9:
		r += "CM"
	case 4:
		r += "CD"
	default:
		if c >= 5 {
			r += "D"
		}
		for i := 0; i < c % 5; i ++ {
			r += "C"
		}
	}

	c, num = num / 10, num % 10
	switch c {
	case 9:
		r += "XC"
	case 4:
		r += "XL"
	default:
		if c >= 5 {
			r += "L"
		}
		for i := 0; i < c % 5; i ++ {
			r += "X"
		}
	}

	c = num
	switch c {
	case 9:
		r += "IX"
	case 4:
		r += "IV"
	default:
		if c >= 5 {
			r += "V"
		}
		for i := 0; i < c % 5; i ++ {
			r += "I"
		}
	}

	return r
}
